integration-tests/tests/aws/resources/session-helper.ts

```diff
@@ -0,0 +1,74 @@
+import { clearItemsFromTables } from "./dynamodb-helper";
+import { StackInfo } from "./cloudformation-helper";
+
+type Input = ReturnType<typeof input>;
+
+export const input = (nino?: string) => ({
```

Comment:
I don't know how I feel about test inputs being put into this `session-helper` file.  Considering these are more util classes, I think the input definitions should stay within the tests themselves. While I understand why you did this and it avoids duplication, I'm not really comfortable with putting this data here. 

For some context, I have tried to do this type of refactoring before but was quickly told we don't want to do this because when it comes to readability of the tests, we can't easily see what the test input is without having to step through the code (in this case, visiting a different file to see the input). Additionally, this is now shared between multiple files, which introduces really tight coupling i.e. if we want to change the inputs for a specific state machine, we might accidentally break other tests. 

Can we keep the test specific inputs within the tests themselves? Or if we want to do this style, have a data file in each of the state machine folders e.g. in `nino-check` have a file containing the inputs, in `check-session` it has its own file etc (but I'd prefer the former of keeping it within the test files themselves).

Reply:
Overall, nice PR, the code looks much cleaner than before.

Reply:
Thanks Suraj, I'll make changes as we've discussed earlier